src/desktopvirtualization/azext_desktopvirtualization/generated/_params.py

```diff
@@ -31,117 +31,155 @@ def load_arguments(self, _):
         c.argument('resource_group_name', resource_group_name_type)
 
     with self.argument_context('desktopvirtualization hostpool show') as c:
         c.argument('resource_group_name', resource_group_name_type)
-        c.argument('host_pool_name', options_list=['--name', '-n'], help='The name of the host pool within the '
-                   'specified resource group', id_part='name')
+        c.argument('host_pool_name', options_list=['--name', '-n'], type=str,
+                   help='The name of the host pool within the specified resource group', id_part='name')
 
     with self.argument_context('desktopvirtualization hostpool create') as c:
         c.argument('resource_group_name', resource_group_name_type)
-        c.argument('host_pool_name', options_list=['--name', '-n'], help='The name of the host pool within the '
-                   'specified resource group')
-        c.argument('tags', tags_type)
-        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+        c.argument('host_pool_name', options_list=['--name', '-n'], type=str,
+                   help='The name of the host pool within the specified resource group')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx), required=False,
                    validator=get_default_location_from_resource_group)
-        c.argument('friendly_name', help='Friendly name of HostPool.')
-        c.argument('description', help='Description of HostPool.')
-        c.argument('host_pool_type', arg_type=get_enum_type(['Personal', 'Pooled']),
+        c.argument('tags', tags_type)
+        c.argument('friendly_name', type=str, help='Friendly name of HostPool.')
+        c.argument('description', type=str, help='Description of HostPool.')
+        c.argument('host_pool_type', arg_type=get_enum_type(['Personal', 'Pooled', 'BYODesktop']),
                    help='HostPool type for desktop.')
-        c.argument('personal_desktop_assignment_type', arg_type=get_enum_type(['Automatic', 'Direct']), help=''
-                   'PersonalDesktopAssignment type for HostPool.')
-        c.argument('custom_rdp_property', help='Custom rdp property of HostPool.')
-        c.argument('max_session_limit', help='The max session limit of HostPool.')
-        c.argument('load_balancer_type', arg_type=get_enum_type(['BreadthFirst', 'DepthFirst', 'Persistent']), help=''
-                   'The type of the load balancer.')
-        c.argument('ring', help='The ring number of HostPool.')
+        c.argument('personal_desktop_assignment_type', arg_type=get_enum_type(['Automatic', 'Direct']),
+                   help='PersonalDesktopAssignment type for HostPool.')
+        c.argument('custom_rdp_property', type=str, help='Custom rdp property of HostPool.')
+        c.argument('max_session_limit', type=int, help='The max session limit of HostPool.')
+        c.argument('load_balancer_type', arg_type=get_enum_type(['BreadthFirst', 'DepthFirst', 'Persistent']),
```

Comment:
Are these enumerations defined in the Python SDK?

Reply:
No, these enumerations only record in python docstrings.